README.md

```diff
@@ -194,6 +194,89 @@ for getting the file list.
 This option cannot be set to `false` with `redirect` set to `true` on a server
 with `ignoreTrailingSlash` set to `true`.
 
+#### `hash`
+
+Default: `undefined`
+
+Enabling `hash` will turn off the `wildcard` option if it wasn't explicitly set to `true`, and both cannot be set to `true`. When enabled, `hash` lets the user access assets dynamically using the decorated `getHashedAsset` function. This in turn makes possible the usage of a very high `maxAge` so that the content can be cached as long as possible. If any modifications are made to a file, its hash will simply be recalculated during the next startup and the cache will bust for that asset.
+
+**Note:** A custom script can be used to generate the hashes in advance to speed up cold start times, ideally during the build phase.
+
+##### Example:
+
+```js
+const fastify = require('fastify')()
+fastify.register(require('fastify-static'), {
+  root: path.join(__dirname, 'public'),
+  hash: true,
+  immutable: true,
+  maxAge: 31536000 // You can set a very long maxAge
+})
+
+fastify.listen(3000, (err) => {
+  if (err) throw err
+  console.log('Server listening at http://localhost:3000')
+})
+```
+
+Then in your templates:
+
+```handlebars
+<script src="{{ fastify.getHashedAsset('js/main.js') }}"></script>
+<link href="{{ fastify.getHashedAsset('css/main.css') }}" rel="stylesheet">
+```
+
+##### Example of a custom script to pregenerate the hashes:
+
+```js
+#!/usr/bin/env node
+
+'use strict'
+
+const { generateHashes } = require('@fastify/static')
+const path = require('path')
+
+async function run () {
+  let [rootPaths, includeDotFiles, ignorePatterns, writeLocation] = process.argv.slice(2)
+
+  if (!rootPaths) {
+    console.error('Usage: hash <root-path> <include-dot-files> <ignore-patterns> <writeLocation>')
+    process.exit(1)
+  }
+
+  rootPaths = rootPaths.split(',').map(p => path.resolve(p.trim()))
+  includeDotFiles = includeDotFiles === 'true'
+  ignorePatterns = ignorePatterns ? ignorePatterns.split(',') : []
+
+  try {
+    await generateHashes(rootPaths, includeDotFiles, ignorePatterns, true, writeLocation)
+    console.log('Hashes generated successfully.')
+  } catch (error) {
+    console.error('Error generating hashes:', error)
+  }
+}
+
+run()
```

Comment:
Why isn't this provided as bin script?

Reply:
Because I just learned what they are, still growing 😁✨

Reply:
Done, called it hash-gen but can be something else as well